### ISSUE TITLE
Cannot send <struct> within MethodCall

### ISSUE THREAD

**Issue by atlantis210** [erickok/retrofit-xmlrpc#6]
Hi,
It's me again :)
I wanted to send a struct field to wordpress (serves to filter the response), but it always sends an empty struct in the MethodCall.
Changing this in the code fixes it. In _ValueConverter.getValue()_.
`else if (value instanceof Map) {
   Map map = (Map) value;
    List<Member> members = new ArrayList<>();
    for (Map.Entry entry : ((Set<Map.Entry>) map.entrySet())) {
       members.add(Member.create(entry.getKey().toString(), ValueConverter.getValue(entry.getValue())));
     }
    return new StructValue(members);
}`
I let you verify it and implement it if you want to.

**Comment by erickok** [comment]
So you tried to set up a map and expected these to end up as the `<struct>` in the request? Yeah that's not how it was designed for the moment, but a totally valid idea. Especially since the reverse - to have `<struct>`s in a response map to a `Map` - is supported already.

**Comment by atlantis210** [comment]
Yes, I wanted to have a `struct` in the MethodCall to add filter for the WordPress query. Furthermore, I removed the `throw exception` in the `ArrayValue.java` file since a field can be empty `if (data.size() != targetFields.size())` and only `return type.newInstance()`.
Anyway, thanks a lot for you library, it saves me a lot of time ! :)
